zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -190,13 +190,13 @@ public static Either<ProblemDetail, UpdateUserTaskRequest> toUserTaskUpdateReque
   }
 
   public static Either<ProblemDetail, UpdateUserRequest> toUserUpdateRequest(
-      final UserUpdateRequest updateRequest, final long userKey) {
+      final UserUpdateRequest updateRequest, final String username) {
     final UserChangeset changeset = updateRequest.getChangeset();
     return getResult(
         validateUserUpdateRequest(updateRequest),
         () ->
             new UpdateUserRequest(
-                userKey, changeset.getName(), changeset.getEmail(), changeset.getPassword()));
+                username, changeset.getName(), changeset.getEmail(), changeset.getPassword()));
```

Comment:
🤔 Is this legacy code? I don't quite understand why we don't return a `UserDTO` here directly, similar to what we do with user creation. Do you know?

Reply:
You mean that we should return a `UserDTO` instance instead of `UpdateUserRequest`?

Reply:
Yeah, I refactored it. We don't need the `UpdateUserRequest` step.